CountsAndLists: Move to Beta Protobuf

Kalaiselvi84 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://github.com/googleforgames/agones/blob/main/CONTRIBUTING.md and developer guide https://github.com/googleforgames/agones/blob/main/build/README.md
2. Please label this pull request according to what type of issue you are addressing.
3. Ensure you have added or ran the appropriate tests for your PR: https://github.com/googleforgames/agones/blob/main/build/README.md#testing-and-building
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Closes #<issue number>`, or `Closes (paste link of issue)`.
-->
Work on #3775 

**Special notes for your reviewer**:
This PR moves all the Counterss & Lists functions from Alpha to Beta protobuf and updates the relevant files accordingly.

Kalaiselvi84 commented on pkg/sdkserver/localsdk.go:
```diff
@@ -42,8 +42,8 @@ import (
 	_ sdk.SDKServer   = &LocalSDKServer{}
-	_ alpha.SDKServer = &LocalSDKServer{}
-	_ beta.SDKServer  = &LocalSDKServer{}
+	_ alpha.SDKServer = &AlphaSDKAdapter{}
```
I was trying to differentiate alpha and beta features. I have removed it.

Kalaiselvi84 commented on cmd/sdk-server/main.go:
```diff
@@ -199,9 +199,12 @@ func registerLocal(grpcServer *grpc.Server, ctlConf config) (func(), error) {
 	sdk.RegisterSDKServer(grpcServer, s)
-	sdkalpha.RegisterSDKServer(grpcServer, s)
-	sdkbeta.RegisterSDKServer(grpcServer, s)
+	sdkalpha.RegisterSDKServer(grpcServer, alphaSDKAdapter)
```
reverted this change

Kalaiselvi84 commented on proto/sdk/alpha/alpha.proto:
```diff
@@ -128,65 +128,6 @@ service SDK {
-      };
-    }
 }
 
```
Deleted the CL related lines

Kalaiselvi84 commented:
`pkg/sdk/alpha/*` and `pkg/sdk/beta/*` files are autogenerated, do we need to execute any command to modify these file?

igooch reviewed (commented):
Looks like the tests are failing `test-gen-all-sdk-grpc`. Run `make gen-all-sdk-grpc` from within the /build directory. Once you run that you'll also need to update the REST test/sdk/restapi/http-api-test.go to use beta.

Kalaiselvi84 commented:
> Looks like the tests are failing `test-gen-all-sdk-grpc`. Run `make gen-all-sdk-grpc` from within the /build directory. Once you run that you'll also need to update the REST test/sdk/restapi/http-api-test.go to use beta.

I ran `make gen-all-sdk-grpc` to modify autogenerated files, updated `build/build-sdk-images/restapi/gen.sh` to automate beta file generation, and adjusted `test/sdk/restapi/http-api-test.go` to change alpha to beta for CL

Kalaiselvi84 commented on proto/sdk/beta/beta.proto:
```diff
@@ -31,4 +37,185 @@ option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
+message Bool {
+    bool bool = 1 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {format: "boolean"}];
+}
+
```
Removed these lines

Kalaiselvi84 commented on test/sdk/go/sdk-client-test.go:
```diff
@@ -160,36 +160,36 @@ func testPlayerTracking(sdk *goSdk.SDK) {
 	// LocalSDKServer starting "rooms": {Count: 1, Capacity: 10}
 	counter := "rooms"
-	count, err := sdk.Alpha().GetCounterCount(counter)
+	count, err := sdk.Beta().GetCounterCount(counter)
```
modified with beta in test/sdk/csharp/Program.cs

Kalaiselvi84 commented:
I will push a new tag for `simple-game-server` to the registry.

igooch reviewed (approved):
LGTM. @markmandel did you want to take a last look?

We still need to update documentation, which can be done as part of a different PR.

zmerlynn commented:
I'm going to go ahead and merge this since it's so wide - @markmandel if you wsnt to follow up we still have the rest of the release. :)

Great job, @Kalaiselvi84 and @igooch !

steven-supersolid commented on sdks/nodejs/lib/alpha/alpha_grpc_pb.js:
Just for posterity but this removed functions from the file and there is no corresponding addition to a `beta_grpc_pb.js` - in fact that file is not generated yet
